### ISSUE TITLE
No sound and no error messages

### ISSUE THREAD

**Issue by ayghor** [jackaudio/jack2#143]
I can't get any sound from jack!

 * running everything as root to ensure I have permissions
 * tested with `fluidsynth -j` and `mpv -ao jack`
 * sound works without jack, but not with jack
 * checked alsamixer and everything is unmuted and has volume
 * there is no pulseaudio installed

Sorry if this is an obvious mistake, I've been searching for a while with no luck.

Version:

    # jackd --version
    jackdmp 1.9.10
    Copyright 2001-2005 Paul Davis and others.
    Copyright 2004-2014 Grame.
    jackdmp comes with ABSOLUTELY NO WARRANTY
    This is free software, and you are welcome to redistribute it
    under certain conditions; see the file COPYING for details
    jackdmp version 1.9.10 tmpdir /dev/shm protocol 8

Output:

    # jackd -d alsa
    jackdmp 1.9.10
    Copyright 2001-2005 Paul Davis and others.
    Copyright 2004-2014 Grame.
    jackdmp comes with ABSOLUTELY NO WARRANTY
    This is free software, and you are welcome to redistribute it
    under certain conditions; see the file COPYING for details
    JACK server starting in realtime mode with priority 10
    self-connect-mode is "Don't restrict self connect requests"
    audio_reservation_init
    Acquire audio card Audio0
    creating alsa driver ... hw:0|hw:0|1024|2|48000|0|0|nomon|swmeter|-|32bit
    configuring for 48000Hz, period = 1024 frames (21.3 ms), buffer = 2 periods
    ALSA: final selected sample format for capture: 32bit integer little-endian
    ALSA: use 2 periods for capture
    ALSA: final selected sample format for playback: 32bit integer little-endian
    ALSA: use 2 periods for playback

Verbose output:

    # jackd -v -d alsa
    jackdmp 1.9.10
    Copyright 2001-2005 Paul Davis and others.
    Copyright 2004-2014 Grame.
    jackdmp comes with ABSOLUTELY NO WARRANTY
    This is free software, and you are welcome to redistribute it
    under certain conditions; see the file COPYING for details
    JACK server starting in realtime mode with priority 10
    self-connect-mode is "Don't restrict self connect requests"
    Jack: JackPosixThread::StartImp : create non RT thread
    Jack: JackPosixThread::ThreadHandler : start
    Jack: JackDriver::Open capture_driver_name = hw:0
    Jack: JackDriver::Open playback_driver_name = hw:0
    Jack: Check protocol client = 8 server = 8
    Jack: JackEngine::ClientInternalOpen: name = system
    Jack: JackEngine::AllocateRefNum ref = 0
    Jack: JackPosixSemaphore::Allocate name = jack_sem.0_default_system val = 0
    Jack: JackEngine::NotifyAddClient: name = system
    Jack: JackGraphManager::SetBufferSize size = 1024
    Jack: JackConnectionManager::DirectConnect first: ref1 = 0 ref2 = 0
    Jack: JackGraphManager::ConnectRefNum cur_index = 0 ref1 = 0 ref2 = 0
    Jack: JackDriver::SetupDriverSync driver sem in flush mode
    audio_reservation_init
    Acquire audio card Audio0
    creating alsa driver ... hw:0|hw:0|1024|2|48000|0|0|nomon|swmeter|-|32bit
    configuring for 48000Hz, period = 1024 frames (21.3 ms), buffer = 2 periods
    ALSA: final selected sample format for capture: 32bit integer little-endian
    ALSA: use 2 periods for capture
    ALSA: final selected sample format for playback: 32bit integer little-endian
    ALSA: use 2 periods for playback
    Jack: JackSocketServerChannel::Open
    Jack: JackServerSocket::Bind : addr.sun_path /dev/shm/jack_default_0_0
    Jack: JackSocketServerChannel::BuildPoolTable size = 1
    Jack: JackEngine::Open
    Jack: JackClientSocket::Connect : addr.sun_path /dev/shm/jack_default_0_0
    Jack: JackEngine::ClientInternalOpen: name = freewheel
    Jack: JackEngine::AllocateRefNum ref = 1
    Jack: JackPosixSemaphore::Allocate name = jack_sem.0_default_freewheel val = 0
    Jack: JackEngine::NotifyAddClient: name = freewheel
    Jack: JackDriver::ClientNotify ref = 1 driver = system name = freewheel notify = 0
    Jack: JackDriver::ClientNotify ref = 0 driver = freewheel name = system notify = 0
    Jack: JackConnectionManager::DirectConnect first: ref1 = 1 ref2 = 1
    Jack: JackGraphManager::ConnectRefNum cur_index = 0 ref1 = 1 ref2 = 1
    Jack: JackDriver::SetupDriverSync driver sem in flush mode
    Jack: JackGraphManager::SetBufferSize size = 1024
    Jack: JackAlsaDriver::Attach fBufferSize 1024 fSampleRate 48000
    Jack: JackEngine::PortRegister ref = 0 name = system:capture_1 type = 32 bit float mono audio flags = 22 buffer_size = 1024
    Jack: JackGraphManager::AllocatePortAux port_index = 1 name = system:capture_1 type = 32 bit float mono audio
    Jack: JackConnectionManager::AddOutputPort ref = 0 port = 1
    Jack: JackEngine::ClientNotify: no callback for notification = 9
    Jack: JackEngine::ClientNotify: no callback for notification = 9
    Jack: JackAlsaDriver::Attach fCapturePortList[i] 1 
    Jack: JackEngine::PortRegister ref = 0 name = system:capture_2 type = 32 bit float mono audio flags = 22 buffer_size = 1024
    Jack: JackGraphManager::AllocatePortAux port_index = 2 name = system:capture_2 type = 32 bit float mono audio
    Jack: JackConnectionManager::AddOutputPort ref = 0 port = 2
    Jack: JackEngine::ClientNotify: no callback for notification = 9
    Jack: JackEngine::ClientNotify: no callback for notification = 9
    Jack: JackAlsaDriver::Attach fCapturePortList[i] 2 
    Jack: JackEngine::PortRegister ref = 0 name = system:playback_1 type = 32 bit float mono audio flags = 21 buffer_size = 1024
    Jack: JackGraphManager::AllocatePortAux port_index = 3 name = system:playback_1 type = 32 bit float mono audio
    Jack: JackConnectionManager::AddInputPort ref = 0 port = 3
    Jack: JackEngine::ClientNotify: no callback for notification = 9
    Jack: JackEngine::ClientNotify: no callback for notification = 9
    Jack: JackAlsaDriver::Attach fPlaybackPortList[i] 3 
    Jack: JackEngine::PortRegister ref = 0 name = system:playback_2 type = 32 bit float mono audio flags = 21 buffer_size = 1024
    Jack: JackGraphManager::AllocatePortAux port_index = 4 name = system:playback_2 type = 32 bit float mono audio
    Jack: JackConnectionManager::AddInputPort ref = 0 port = 4
    Jack: JackEngine::ClientNotify: no callback for notification = 9
    Jack: JackEngine::ClientNotify: no callback for notification = 9
    Jack: JackAlsaDriver::Attach fPlaybackPortList[i] 4 
    Jack: JackEngine::PortRegister ref = 0 name = system:playback_3 type = 32 bit float mono audio flags = 21 buffer_size = 1024
    Jack: JackGraphManager::AllocatePortAux port_index = 5 name = system:playback_3 type = 32 bit float mono audio
    Jack: JackConnectionManager::AddInputPort ref = 0 port = 5
    Jack: JackEngine::ClientNotify: no callback for notification = 9
    Jack: JackEngine::ClientNotify: no callback for notification = 9
    Jack: JackAlsaDriver::Attach fPlaybackPortList[i] 5 
    Jack: JackEngine::PortRegister ref = 0 name = system:playback_4 type = 32 bit float mono audio flags = 21 buffer_size = 1024
    Jack: JackGraphManager::AllocatePortAux port_index = 6 name = system:playback_4 type = 32 bit float mono audio
    Jack: JackConnectionManager::AddInputPort ref = 0 port = 6
    Jack: JackEngine::ClientNotify: no callback for notification = 9
    Jack: JackEngine::ClientNotify: no callback for notification = 9
    Jack: JackAlsaDriver::Attach fPlaybackPortList[i] 6 
    Jack: JackEngine::PortRegister ref = 0 name = system:playback_5 type = 32 bit float mono audio flags = 21 buffer_size = 1024
    Jack: JackGraphManager::AllocatePortAux port_index = 7 name = system:playback_5 type = 32 bit float mono audio
    Jack: JackConnectionManager::AddInputPort ref = 0 port = 7
    Jack: JackEngine::ClientNotify: no callback for notification = 9
    Jack: JackEngine::ClientNotify: no callback for notification = 9
    Jack: JackAlsaDriver::Attach fPlaybackPortList[i] 7 
    Jack: JackEngine::PortRegister ref = 0 name = system:playback_6 type = 32 bit float mono audio flags = 21 buffer_size = 1024
    Jack: JackGraphManager::AllocatePortAux port_index = 8 name = system:playback_6 type = 32 bit float mono audio
    Jack: JackConnectionManager::AddInputPort ref = 0 port = 8
    Jack: JackEngine::ClientNotify: no callback for notification = 9
    Jack: JackEngine::ClientNotify: no callback for notification = 9
    Jack: JackAlsaDriver::Attach fPlaybackPortList[i] 8 
    Jack: JackEngine::PortRegister ref = 0 name = system:playback_7 type = 32 bit float mono audio flags = 21 buffer_size = 1024
    Jack: JackGraphManager::AllocatePortAux port_index = 9 name = system:playback_7 type = 32 bit float mono audio
    Jack: JackConnectionManager::AddInputPort ref = 0 port = 9
    Jack: JackEngine::ClientNotify: no callback for notification = 9
    Jack: JackEngine::ClientNotify: no callback for notification = 9
    Jack: JackAlsaDriver::Attach fPlaybackPortList[i] 9 
    Jack: JackEngine::PortRegister ref = 0 name = system:playback_8 type = 32 bit float mono audio flags = 21 buffer_size = 1024
    Jack: JackGraphManager::AllocatePortAux port_index = 10 name = system:playback_8 type = 32 bit float mono audio
    Jack: JackConnectionManager::AddInputPort ref = 0 port = 10
    Jack: JackEngine::ClientNotify: no callback for notification = 9
    Jack: JackEngine::ClientNotify: no callback for notification = 9
    Jack: JackAlsaDriver::Attach fPlaybackPortList[i] 10 
    Jack: Clock source : system clock via clock_gettime
    Jack: JackServer::Start
    Jack: JackThreadedDriver::Start
    Jack: JackPosixThread::StartImp : create non RT thread
    Jack: JackPosixThread::ThreadHandler : start
    Jack: JackThreadedDriver::Init real-time
    Jack: JackPosixThread::AcquireRealTimeImp priority = 10
    Jack: JackPosixThread::StartImp : create non RT thread
    Jack: JackPosixThread::ThreadHandler : start
    Jack: JackSocketServerChannel::ClientCreate socket
    Jack: JackSocketServerChannel::BuildPoolTable size = 2
    Jack: JackSocketServerChannel::BuildPoolTable fSocketTable i = 1 fd = 9
    Jack: JackSocketServerChannel::Execute : fPollTable i = 1 fd = 9
    Jack: JackRequest::Notification
    Jack: JackDriver::ClientNotify ref = 1 driver = freewheel name = freewheel notify = 18
    Jack: JackDriver::ClientNotify ref = 1 driver = freewheel name = freewheel notify = 18
    Jack: JackDriver::ClientNotify ref = 1 driver = freewheel name = freewheel notify = 18
    Jack: JackDriver::ClientNotify ref = 1 driver = freewheel name = freewheel notify = 18
    Jack: JackEngine::ClientNotify: no callback for notification = 4
    Jack: JackEngine::ClientNotify: no callback for notification = 4

**Comment by lilrc** [comment]
I have no idea what might be wrong, but let's start somewhere. What does `aplay -lL` give you? Have you tried different samplerates? When the jack server and any audio program, e.g. `fluidsynth`, is running, what does `jack_lsp -c` give you? Can you try another program such as `jack_metro` and see if it works?

By the way, which distribution are you on? I'm curious to hear which distros (other than Gentoo) that do not force people to use PulseAudio.

**Comment by ayghor** [comment]
Sorry for taking so long!

`aplay -lL` gives

    null
        Discard all samples (playback) or generate zero samples (capture)
    default:CARD=Intel
        HDA Intel, AD1988B Analog
        Default Audio Device
    sysdefault:CARD=Intel
        HDA Intel, AD1988B Analog
        Default Audio Device
    front:CARD=Intel,DEV=0
        HDA Intel, AD1988B Analog
        Front speakers
    surround21:CARD=Intel,DEV=0
        HDA Intel, AD1988B Analog
        2.1 Surround output to Front and Subwoofer speakers
    surround40:CARD=Intel,DEV=0
        HDA Intel, AD1988B Analog
        4.0 Surround output to Front and Rear speakers
    surround41:CARD=Intel,DEV=0
        HDA Intel, AD1988B Analog
        4.1 Surround output to Front, Rear and Subwoofer speakers
    surround50:CARD=Intel,DEV=0
        HDA Intel, AD1988B Analog
        5.0 Surround output to Front, Center and Rear speakers
    surround51:CARD=Intel,DEV=0
        HDA Intel, AD1988B Analog
        5.1 Surround output to Front, Center, Rear and Subwoofer speakers
    surround71:CARD=Intel,DEV=0
        HDA Intel, AD1988B Analog
        7.1 Surround output to Front, Center, Side, Rear and Woofer speakers
    iec958:CARD=Intel,DEV=0
        HDA Intel, AD1988B Digital
        IEC958 (S/PDIF) Digital Audio Output
    hdmi:CARD=NVidia,DEV=0
        HDA NVidia, HDMI 0
        HDMI Audio Output
    hdmi:CARD=NVidia,DEV=1
        HDA NVidia, HDMI 0
        HDMI Audio Output
    hdmi:CARD=NVidia,DEV=2
        HDA NVidia, HDMI 0
        HDMI Audio Output
    hdmi:CARD=NVidia,DEV=3
        HDA NVidia, HDMI 0
        HDMI Audio Output
    **** List of PLAYBACK Hardware Devices ****
    card 0: Intel [HDA Intel], device 0: AD1988B Analog [AD1988B Analog]
      Subdevices: 1/1
      Subdevice #0: subdevice #0
    card 0: Intel [HDA Intel], device 1: AD1988B Digital [AD1988B Digital]
      Subdevices: 1/1
      Subdevice #0: subdevice #0
    card 0: Intel [HDA Intel], device 2: AD1988B Alt Analog [AD1988B Alt Analog]
      Subdevices: 1/1
      Subdevice #0: subdevice #0
    card 1: NVidia [HDA NVidia], device 3: HDMI 0 [HDMI 0]
      Subdevices: 1/1
      Subdevice #0: subdevice #0
    card 1: NVidia [HDA NVidia], device 7: HDMI 0 [HDMI 0]
      Subdevices: 1/1
      Subdevice #0: subdevice #0
    card 1: NVidia [HDA NVidia], device 8: HDMI 0 [HDMI 0]
      Subdevices: 1/1
      Subdevice #0: subdevice #0
    card 1: NVidia [HDA NVidia], device 9: HDMI 0 [HDMI 0]
      Subdevices: 1/1
      Subdevice #0: subdevice #0

`jack_lsp -c` while running `fluidsynth -j` gives

    system:capture_1
    system:capture_2
    system:playback_1
       fluidsynth:l_00
    system:playback_2
       fluidsynth:r_00
    system:playback_3
    system:playback_4
    system:playback_5
    system:playback_6
    system:playback_7
    system:playback_8
    fluidsynth:l_00
       system:playback_1
    fluidsynth:r_00
       system:playback_2

`jack_metro -b 200` gives no sound and no output. (I ended fluidsynth
first!)

`jack_lsp -c` while running `jack_metro -b 200` gives

    system:capture_1
    system:capture_2
    system:playback_1
    system:playback_2
    system:playback_3
    system:playback_4
    system:playback_5
    system:playback_6
    system:playback_7
    system:playback_8
    metro:200_bpm
    
My distro is void-linux: http://www.voidlinux.eu/

And I just noticed one thing: jack was compiled with pulseaudio libs,
even though it's executables are not a dependency to teh package and I
don't have them installed. When I run jack, it consumes my sound card,
muting firefox, for example. Could it still be consuming teh resource
and waiting for pulseaudio? I did `grep -i pulse` with no results, guess
it should at least mention it somewhere.

**Comment by ayghor** [comment]
Sorry for this, I didn't pay much attention to teh output of `jack_lsp -c`. Teh second time I ran it, it had no connections up. Then I tried to connect evey port to teh metronome and it did beep on `system:playback_5`.

I feel stupid. Back when it was working, some time ago, it worked on ports 1 and 2. But for some reason, this changed and I didn't pay attention.

Also: Shouldn't there be two ports working, one for Left and one for Right channel?
